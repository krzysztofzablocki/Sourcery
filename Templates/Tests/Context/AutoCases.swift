//
//  AutoCases.swift
//  Templates
//
//  Created by Anton Domashnev on 03.05.17.
//  Copyright Â© 2017 Pixle. All rights reserved.
//

import Foundation

protocol AutoCases {}

enum AutoCasesEnum: AutoCases {
    case north
    case south
    case east
    case west
}

enum AutoCasesOneValueEnum: AutoCases {
    case one
}

public enum AutoCasesHasAssociatedValuesEnum: AutoCases {
    case foo(test: String)
    case bar(number: Int)
}
