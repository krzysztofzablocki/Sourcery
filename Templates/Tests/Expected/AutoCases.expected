internal extension AutoCasesEnum {
  static let count: Int = 4
  static let allCases: [AutoCasesEnum] = [
    .north,
    .south,
    .east,
    .west
  ]
}
public extension AutoCasesHasAssociatedValuesEnum {
  static let count: Int = 2
}
internal extension AutoCasesOneValueEnum {
  static let count: Int = 1
  static let allCases: [AutoCasesOneValueEnum] = [
    .one
  ]
}

